### PR TITLE
Add eslint

### PR DESCRIPTION
**Pull request by WiNloSt**
This PR add ESLint to the project using `npm init @eslint/config@latest` command. With some manual alteration to the config to suit our project.

And use `husky` + `lint-staged` to automatically check JS/TS files with ESLint.

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      "react/react-in-jsx-scope": "off",
+      "@typescript-eslint/no-unused-vars": [
+        "error",
+        {
+          argsIgnorePattern: "^_",
+        },
+      ],
+      "react/prop-types": "warn",
+    },
+  },
```
Comment (WiNloSt) [inline eslint.config.js:25]: I only added this section. The rest are generated.

File: src/components/ThemeProvider.tsx
Comment (WiNloSt) [inline src/components/ThemeProvider.tsx:1]: remove unused vars

File: src/routes/analytics/AnalyticsOverviewPage.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ export function AnalyticsOverviewPage() {
 
         <SimpleGrid cols={{ base: 2, sm: 3, lg: 4 }}>
           {overviewLinkCards.map((card) => (
-            <DashboardLinkCard {...card} />
+            <DashboardLinkCard key={card.id} {...card} />
```
Comment (WiNloSt) [inline src/routes/analytics/AnalyticsOverviewPage.tsx:19]: require `key`

File: package.json [hunk 3]
```diff
@@ -66,12 +68,20 @@
   },
   "devDependencies": {
     "@babel/plugin-proposal-private-property-in-object": "^7.21.11",
+    "@eslint/compat": "^1.1.0",
```
Comment (WiNloSt) [inline package.json:71]: all ESLint related dependencies are added via `npm init @eslint/config@latest` command.

File: package.json [hunk 4]
```diff
@@ -44,12 +44,14 @@
...
+    "lint": "eslint src",
+    "prepare": "husky"
   },
-  "eslintConfig": {
-    "extends": [
-      "react-app",
-      "react-app/jest"
+  "lint-staged": {
+    "*.{js,jsx,ts,tsx}": [
+      "yarn prettier --write",
```
Comment (npretto) [inline package.json:53]: should we run prettier later?
I think there are eslint rules that may add/remove code, in the main repo we have first eslint and then prettier

Comment (WiNloSt) [reply package.json]: Fixed in https://github.com/metabase/shoppy/pull/48/commits/6db3e92dda5dd90606781d10acad023876666a7b

**Review by heypoom — APPROVED** [review]
ESLint config looks good to me!

**Pull request merged**